core/src/main/java/org/frankframework/filesystem/SftpFileSystem.java

```diff
@@ -66,10 +69,36 @@ public void close() {
 		ftpClient = null;
 	}
 
-
 	@Override
 	public boolean isOpen() {
-		return ftpClient != null && ftpClient.isConnected();
+		log.debug("checking if SFTP connection is open");
+		try {
+			if (ftpClient == null || ftpClient.getSession() == null) {
+				return false;
+			}
+			ftpClient.getSession().sendKeepAliveMsg(); // Send a keep-alive packet to validate a working connection
+		} catch (Exception ignored) {
+			return false;
+		}
+		return ftpClient.isConnected();
+	}
+
+	private void reconnectWhenNotConnected() {
+		if (lastCheck + RECHECK_CONNECTION_INTERVAL_SECONDS * 1000L > System.currentTimeMillis()) {
+			return;
+		}
+		if (!isOpen()) {
+			try {
+				log.info("Reconnecting to SFTP server, since connection was closed");
+				if (ftpClient == null) {
+					open();
+				}
```

Comment:
Wie zou hem moeten sluiten? Ik denk dat dit niet nodig is en hij hier gewoon false moet teruggeven.

Reply:
Ja, inderdaad, terechte opmerking. Ik ben er nu nog effe mee bezig. Met een verse blik zie ik dat het anders moet.